internal/edge/logs.go

```diff
@@ -25,8 +25,8 @@ const (
 	logFailed
 )
 
-func newLogsManager(portainerURL, endpointID, edgeID string, insecurePoll bool) *logsManager {
-	cli := client.NewPortainerClient(portainerURL, endpointID, edgeID, insecurePoll)
+func newLogsManager(portainerURL, endpointID, edgeID string, insecurePoll bool, tunnel bool) *logsManager {
+	cli := client.NewPortainerClient(portainerURL, endpointID, edgeID, insecurePoll, tunnel)
```

Comment:
@deviantony this is literally what we were talking about wrt https://github.com/portainer/agent/pull/234/files#diff-03cd554fc7298e6b35b7371c9c421c6929a3f169b5698f0206ec261d99677446R29

this new `tunnel` boolean is only used at startup, to create something - and so has no reason to be stored in the 3 structs in poll.go, and also passed into what are 2 constructors.

i don't know if its better to merge this, so @joshua-portainer can keep moving, or if doing a little clean up to reduce the baggage of storing singleton construction first is better tho

Reply:
@SvenDowideit What about merging it to keep moving and I'll take care of opening a separate PR to do a cleanup? This will also allow me to review the approach you used with the HTTP client and apply it to that specific case so we can review it afterwards.